exoscale/client.go

```diff
@@ -47,7 +47,8 @@ type BaseConfig struct {
 }
 
 func getClient(endpoint string, meta interface{}) *egoscale.Client {
-	config := meta.(BaseConfig)
+	t := meta.(map[string]interface{})
+	config := t["config"].(BaseConfig)
```

Comment:
These two lines repeat in four places. We could put them into a function e.g. getConfig().